### PR TITLE
BUG-1990467: WIP - Don't run deletion worker with KubeAPI is on

### PR DESCRIPTION
**Pull request by nmagnezi**
# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by nmagnezi** [comment]
/hold
this partially reverts https://github.com/openshift/assisted-service/pull/1801, for the reason [mentioned here](https://bugzilla.redhat.com/show_bug.cgi?id=1990467#c2).
At this point, I want to see the CI test results.

**Comment by nmagnezi** [comment]
@masayag, @filanov: a question about https://github.com/openshift/assisted-service/pull/1801,
I see that with KubeAPI `Deregister Worker` is not invoked, but `Deletion Worker` is. Why is that?

File: cmd/main.go [hunk 1]
```diff
@@ -357,11 +357,11 @@ func main() {
 		crdUtils = controllers.NewDummyCRDUtils()
 	}
 
-	if Options.EnableDeregisterInactiveGC || Options.EnableDeletedUnregisteredGC {
+	if !Options.EnableKubeAPI && (Options.EnableDeregisterInactiveGC || Options.EnableDeletedUnregisteredGC) {
```
Comment (filanov) [inline cmd/main.go:360]: `EnableDeletedUnregisteredGC` should probably be enabled even with kube-api, we will need to clear old deleted clusters from the DB

**Comment by filanov** [comment]
> @masayag, @filanov: a question about #1801,
> I see that with KubeAPI `Deregister Worker` is not invoked, but `Deletion Worker` is. Why is that?

As far as i know deleted worker remove hosts and cluster from the DB, when user delete cluster/hosts then they are just marked as deleted and not removed from the db. we are using it to get info about clusters even if users deleted them.
This worker will delete them from the DB after 2 weeks

**Comment by nmagnezi** [comment]
@filanov thanks.
As we discussed, I closed that bug.
https://bugzilla.redhat.com/show_bug.cgi?id=1990467#c4

File: cmd/main.go (hunk 1 above)
Comment (masayag) [reply cmd/main.go:360]: @filanov back then we agreed that the deregister-inactive-cluster shouldn't be triggered for kubeapi, however, if CRD & services can be used on the same cluster, we should change this and always enable the worker.
The needed change will be in the [query for the clusters from the DB ](https://github.com/openshift/assisted-service/blob/7088bc266b5953eae1764afdd5f6c373963e36cd/internal/cluster/cluster.go#L1048) not to retrieve clusters that has the kube_key_name and kube_key_namespace (those need to be controlled by the lifecycle of kube_key_namespace/kube_key ClusterDeployment)

**Pull request closed without merging**